File: pinyougou-parent/pinyougou-cart-interface/src/main/java/com/cynen/cart/service/CartService.java
package com.cynen.cart.service;

import java.util.List;

import com.cynen.dto.Cart;

public interface CartService {
	
	
	/**
	 * 添加商品到购物车
	 * @param cartList
	 * @param itemId
	 * @param num
	 * @return
	 */
	public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
	
	/**
	 * 从redis中读取指定用户的购物车
	 * @param username
	 * @return
	 */
	public List<Cart> findCartListFromRedis(String username);
	/**
	 * 将当前用户的购物车,保存到Redis中.
	 * @param username
	 * @param cartList
	 */
	public void saveCartListToRedis(String username,List<Cart> cartList);
}

File: pinyougou-parent/pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function($http){
	

	// 查询列表.
	this.findCartList=function(){
		return $http.get("../cart/findCartList.do");
	}
	
	// 添加购物车.
	this.addGoodsToCart=function(itemId,num){
		return $http.get("../cart/addGoodsToCart.do?itemId="+itemId+"&num="+num);
	}
	
	// 求合计数和总金额.
	this.sum=function(cartList){
		var totalValue={totalNum:0,totalMoney:0}
		// 遍历购物车
		for(var i=0;i<cartList.length;i++){
			// 遍历购物车对象
			for(var j=0;j<cartList[i].orderItemList.length;j++){
				totalValue.totalNum+=cartList[i].orderItemList[j].num;
				totalValue.totalMoney+=cartList[i].orderItemList[j].totalFee;
			}
		}
		return totalValue;
	}
	
});